wasm.md

```diff
@@ -589,72 +590,101 @@ The `*_local` instructions are defined here.
 Function Declaration and Invocation
 -----------------------------------
 
-### Function Declaration
-
-Function declarations can look quite different depending on which fields are ommitted and what the context is.
-Here, we allow for an "abstract" function declaration using syntax `func_::___`, and a more concrete one which allows arbitrary order of declaration of parameters, locals, and results.
+### Function Types Gathering
 
 ```k
     syntax TypeKeyWord ::= "param" | "result" | "local"
  // ---------------------------------------------------
 
     syntax FuncDecl  ::= "(" FuncDecl ")"     [bracket]
                        | TypeKeyWord ValTypes
-                       | "export" Identifier
     syntax FuncDecls ::= List{FuncDecl, ""} [klabel(listFuncDecl)]
  // --------------------------------------------------------------
 
-    syntax Instr ::= "(" "func"              FuncDecls Instrs ")"
-                   | "(" "func" Identifier FuncDecls Instrs ")"
-                   | "func" Identifier "::" FuncType VecType "{" Instrs "}"
- // -----------------------------------------------------------------------
-    rule <k> ( func FDECLS INSTRS )
-          => func gatherExportedName(FDECLS) :: gatherFuncType(FDECLS) gatherTypes(local, FDECLS) { INSTRS }
-         ...
+    syntax FuncType ::= gatherFuncType ( FuncDecls ) [function]
+ // -----------------------------------------------------------
+    rule gatherFuncType(FDECLS) => gatherTypes(param, FDECLS) -> gatherTypes(result, FDECLS)
+
+    syntax VecType ::=  gatherTypes ( TypeKeyWord , FuncDecls )            [function]
+                     | #gatherTypes ( TypeKeyWord , FuncDecls , ValTypes ) [function]
+ // ---------------------------------------------------------------------------------
+    rule gatherTypes(TKW, FDECLS) => #gatherTypes(TKW, FDECLS, .ValTypes)
+
+    rule #gatherTypes(TKW , .FuncDecls            , TYPES) => [ TYPES ]
+    rule #gatherTypes(TKW , FDECL:FuncDecl FDECLS , TYPES) => #gatherTypes(TKW, FDECLS, TYPES) [owise]
+
+    rule #gatherTypes(TKW , TKW VTYPES' FDECLS:FuncDecls , VTYPES)
+      => #gatherTypes(TKW ,             FDECLS           , VTYPES + VTYPES')
+```
+
+### Function Export Definition
+
+This section introduce how we expand the folded form of function export.
+Currently, in the expanded form, the `export`s will come after the definition of function.
+`#asId ( Int )` is used to generate a fresh identifier.
+
+```k
+    syntax Identifier  ::= #asId ( Int )
```

Comment:
Let's declare this in `data.md` with the other `Identifier`. And maybe name it `#fresh` instead of `#asId`.